evalml/pipelines/pipeline_base.py

```diff
@@ -477,7 +482,7 @@ def __eq__(self, other):
         random_state_eq = check_random_state_equality(self.random_state, other.random_state)
         if not random_state_eq:
             return False
-        attributes_to_check = ['parameters', '_is_fitted', 'component_graph', 'input_feature_names']
+        attributes_to_check = ['parameters', '_is_fitted', 'component_graph', 'input_feature_names', 'input_target_name']
```

Comment:
I think it would be helpful to add a case where the input target name is different in  `test_pipeline_equality`